labs/a5.md

```diff
@@ -0,0 +1,336 @@
+2. Describe the `tcp_syncookies` `sysctl` option.  How can we toggle this value on, and when would we want this on?
+3. Where does `dhclient` store old leases?
+4. Please write a few stanzas that configures an interface called `test` and configures it so that it is brought up on boot and given the following address: `192.168.13.37/16`.  Where should this snippet be placed?
+5. Assume the following information:
+* **/etc/resolv.conf** file has 3 `nameserver` entries and a `options timeout:1` entry.
+* A successful DNS response takes 20 ms.
+
+  You need to add the `retry:n` option so that you retry a query as many times as possible but the total time to resolve a name, irrelevant of success or failure, takes less than **5** seconds.  What should the value of n be?
+6. If we preferred name resolution be done dynamically rather than using static entries in **/etc/hosts** what file do we need to edit and what is the line we should add?
+
+## :fire: This is fine :fire:
+This section will have you thinking like a sysadmin.  
+
+Each script **might** make changes to your network stack with the intent of damaging your machine's connectivity.  To confine the scope of the 'attacks', scripts will specifically try to alter your connectivity to `google.com` and `ocf.berkeley.edu`.  These scripts are **dangerous** so we are having you run them inside a disposable vm which you should install using the following steps.
+
+1. Install `vagrant` by either downloading it [here][vagrant-download] or using the appropriate package manager for your system.
+2. Install `virtualbox` by either downloading it [here][vbox-download] or using the appropriate package manager for your system.
+  
+   As a side note the installation and configuration scripts have been tested with  `vagrant 2.0.2` and `virtualbox 5.2.8`.  They have additionally been confirmed to work with `vagrant v1.9.x` and `virtualbox 5.1.34`.  Compatibility issues may arise with other versions, please try shifting to the above listed versions or come to office hours.
+   
+3. On your own local machine i.e. laptop or desktop, clone the repo found [here][decal-labs].
+4. Go into the vm directory `cd decal-labs/a5/vm`.
+5. Run `vagrant up`.  Vagrant will automatically bring up a virtual machine on your local machine according to the files in the `vm` directory.
+6. `ssh` into your machine by running `ssh vagrant@192.168.42.42`, if this prompts you for a password it should default to `vagrant`.
+7. Your vm should be configured and also have the `decal-labs` repo cloned to your home directory.  Scripts are numbered and located under `decal-labs/a5/scenarios`.
+8. **Launch scripts, one at a time, with sudo, i.e. `sudo python3 <script.py>`.**
+9. For each script. You will be doing a 2 step process.  **Only move onto another script, once you have finished resolving your current one.**
```

Comment:
nit: `For each script. You will be doing a 2 step process.` -> `For each script, follow this two step process:`

Reply:
fixed